Cython/Compiler/Symtab.py

```diff
@@ -493,17 +493,35 @@ def declare(self, name, cname, type, pos, visibility, shadow = 0, is_type = 0, c
             warning(pos, "'%s' is a reserved name in C." % cname, -1)
 
         entries = self.entries
+        entry = None
+        cpp_already_overridden = False
         if name and name in entries and not shadow and not self.is_builtin_scope:
             old_entry = entries[name]
```

Comment:
I've proposed an update for this section in https://github.com/cython/cython/commit/a440bd688b45a40919d76be50c167d253ff248b6.

I've tried very hard not to change your underlying logic. However, it's making "declare" quite complicated.

Instead I'm arguing that most of this can be moved into `CppClassScope` leaving the base class declare to be relatively simple. Let me know if that seems reasonable - feel free to modify it.